### PR TITLE
store/tikv:move definition of TiKVClient to store/tikv

### PR DESCRIPTION
**Pull request by AndreMouche**
Signed-off-by: shirly <AndreMouche@126.com>

### What problem does this PR solve?

Issue Number: part of #22513 

Problem Summary:
store/tikv: move definition of `TiKVClient` to store/tikv

### What is changed and how it works?

What's Changed:
store/tikv: move definition of `TiKVClient` to store/tikv

### Check List 

Tests 

- Unit test

### Release note 

- No release note

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -61,8 +60,6 @@ const (
 	DefHost = "0.0.0.0"
 	// DefStatusHost is the default status host of TiDB
 	DefStatusHost = "0.0.0.0"
-	// DefStoreLivenessTimeout is the default value for store liveness timeout.
-	DefStoreLivenessTimeout = "5s"
```
Comment (disksing) [inline config/config.go:65]: Do we have backward compatibility issue here?

Comment (AndreMouche) [reply config/config.go:65]: nop

**Review by andylokandy — APPROVED** [review]

**Review by disksing — APPROVED** [review]

**Pull request merged**